test/test_mcp.py

```diff
@@ -23,12 +23,26 @@ def test_gradio_mcp_server_initialization():
 
 def test_get_block_fn_from_tool_name():
     server = GradioMCPServer(app)
-    result = server.get_block_fn_from_tool_name("test_tool")
+    result = server.get_block_fn_from_endpoint_name("test_tool")
     assert result == app.fns[0]
-    result = server.get_block_fn_from_tool_name("nonexistent_tool")
+    result = server.get_block_fn_from_endpoint_name("nonexistent_tool")
     assert result is None
 
 
+def test_generate_tool_names_correctly_for_interfaces():
+    def echo(x):
```

Comment:
Maybe test a lambda function and class method as well

Reply:
added